src/package/subscriberPackageVersion.ts

```diff
@@ -0,0 +1,503 @@
+    connection: Connection,
+    packageInstallRequestOrId: string | PackagingSObjects.PackageInstallRequest,
+    installationKey?: string,
+    options?: PackageInstallOptions
+  ): Promise<PackagingSObjects.PackageInstallRequest> {
+    const id = typeof packageInstallRequestOrId === 'string' ? packageInstallRequestOrId : packageInstallRequestOrId.Id;
+    const packageInstallRequest = await getStatus(connection, id);
+    const pollingTimeout = numberToDuration(options.pollingTimeout) ?? Duration.milliseconds(0);
+    if (pollingTimeout.milliseconds <= 0) {
+      return packageInstallRequest;
+    } else {
+      await waitForPublish(
+        connection,
+        packageInstallRequest.SubscriberPackageVersionKey,
+        options?.publishFrequency || 0,
+        options?.publishTimeout || 0,
+        installationKey
+      );
+      return pollStatus(connection, id, options);
+    }
+  }
+
+  /**
+   * Reports on the progress of a package version uninstall.
+   *
+   * @param id the 06y package version uninstall request id
+   * @param connection
+   */
+  public static async uninstallStatus(
+    id: string,
+    connection: Connection
+  ): Promise<PackagingSObjects.SubscriberPackageVersionUninstallRequest> {
+    if (!id.startsWith('06y') || !sfdc.validateSalesforceId(id)) {
+      throw messages.createError('packageVersionUninstallRequestIdInvalid', [id]);
+    }
+    const result = (await connection.tooling.retrieve(
+      'SubscriberPackageVersionUninstallRequest',
+      id
+    )) as PackagingSObjects.SubscriberPackageVersionUninstallRequest;
+    if (result.Status === 'Error') {
+      const errorDetails = await getUninstallErrors(connection, id);
+      const errors = errorDetails.map((record, index) => `(${index + 1}) ${record.Message}`);
+      const errHeader = errors.length > 0 ? `\n=== Errors\n${errors.join('\n')}` : '';
+      const err = messages.getMessage('defaultErrorMessage', [id, result.Id]);
+
+      throw new SfError(`${err}${errHeader}`, 'UNINSTALL_ERROR', [messages.getMessage('action')]);
+    }
+    return result;
+  }
+  /**
+   * Retrieves the package version create request.
+   *
+   * @param installRequestId
+   * @param connection
+   */
+  public static async getInstallRequest(
+    installRequestId: string,
+    connection: Connection
+  ): Promise<PackagingSObjects.PackageInstallRequest> {
+    if (!installRequestId.startsWith('0Hf') || !sfdc.validateSalesforceId(installRequestId)) {
+      throw messages.createError('packageVersionInstallRequestIdInvalid', [installRequestId]);
+    }
+    const installRequest = await getStatus(connection, installRequestId);
+    if (!installRequest) {
+      throw messages.createError('packageVersionInstallRequestNotFound', [installRequestId]);
+    }
+    return installRequest;
+  }
+
+  /**
+   * Get the package version ID for this PackageVersion.
+   *
+   * @returns The PackageVersionId (05i).
+   */
+  public async getId(): Promise<string> {
```

Comment:
why is this async?

Reply:
Does not need to be, but I will retain the promise return type for consistency across getters.

Reply:
ok, it's breaking an eslint rule, but I'll leave it be

Reply:
I removed async and return a Promise.resolve...